lib/tasks/ahoy_activity_rollup.rake

```diff
@@ -0,0 +1,79 @@
+namespace :fromthepage do
+  desc "Daily Task to Rollup Ahoy Events into Transcribe minutes by Date, User, and Collection"
+  task ahoy_activity_rollup: :environment do
+    rollup_transcribe_for_date()
+  end
+
+  desc "An ad-hoc task to populate the ahoy rollup for historical data"
+  task ahoy_rollup_crawl: :environment do
+
+    ## Figure out how long we need to count back
+    first_ahoy_event = Date.new(2017, 10, 22).beginning_of_day
+    days = ( (Time.now.beginning_of_day - first_ahoy_event) / 60 / 60 / 24).to_i
+    
+    
+    # Count back from yesterday into the past
+    (1..days).each do |n|
+      date = n.days.ago
+
+      print "\n-- Ahoy Rollup #{date.strftime("%Y-%m-%d")} --\n"
+
+      # Perform the rollup for the day
+      rollup_transcribe_for_date(date)
+    end
+  end
+
+  def rollup_transcribe_for_date(day=1.day.ago)
```

Comment:
I generally like to move business logic (even for recurring tasks) out of the body of rake tasks and into either static methods on a model or onto a separate library.   That lets the rake task focus on argument parsing and user notification, and encapsulates the logic to a place where it can be reused by the application or different rake tasks.  See `lib/contentdm_translator.rb` and its relationship to the two `lib/tasks/cdm_*` tasks for an example.